framework/core/js/src/common/helpers/avatar.tsx

```diff
@@ -31,12 +37,24 @@ export default function avatar(user: User | null, attrs: ComponentAttrs = {}): M
 
     if (hasTitle) attrs.title = attrs.title || username;
 
+    // If alt has not been explicitly set, default it to the username
+    // so screen readers have meaningful context.
+    if (attrs.alt === undefined) {
+      attrs.alt = username;
+    }
+
     if (avatarUrl) {
-      return <img {...attrs} src={avatarUrl} alt="" />;
+      return <img {...attrs} src={avatarUrl} />;
     }
 
     content = username.charAt(0).toUpperCase();
     attrs.style = { '--avatar-bg': user.color() };
```

Comment:
_Don't commit the following thru GH, the diff kind of got messed up_

```suggestion
content = username.charAt(0).toUpperCase();
attrs.style = { '--avatar-bg': user.color() };

delete attrs.alt;
delete attrs.loading;
attrs.role = 'img';
attrs['aria-label'] = username;
```

When no avatarUrl exists, this extra logic removes the alt (and for good measures the `loading="lazy"`  property) and instead sets the aria `role` to `img` and the `aria-label` to username

Reply:
I did that in #4243 not sure why I didn't do it here! 🙈